src/Compilers/CSharp/Portable/Symbols/Source/ModifierUtils.cs

```diff
@@ -245,8 +245,26 @@ internal static void CheckFeatureAvailabilityForPartialEventsAndConstructors(Loc
         }
 #nullable disable
 
-        internal static DeclarationModifiers AdjustModifiersForAnInterfaceMember(DeclarationModifiers mods, bool hasBody, bool isExplicitInterfaceImplementation)
+        internal static DeclarationModifiers AdjustModifiersForAnInterfaceMember(DeclarationModifiers mods, bool hasBody, bool isExplicitInterfaceImplementation, bool forMethod, CSharpSyntaxNode syntaxNode)
         {
+            // In C# 13 and earlier, interface partial members were not implicitly public and virtual and that is fixed only in C# 14 and later to avoid a breaking change.
+            bool newPartialBehavior = (mods & DeclarationModifiers.Partial) == 0 || ((CSharpParseOptions)syntaxNode.SyntaxTree.Options).LanguageVersion > LanguageVersion.CSharp13;
```

Comment:
>newPartialBehavior [](http://example.com/codeflow?start=17&length=18)

Setting this to `true` regardless of language version feels unintuitive and confusing.

Reply:
Perhaps changing the name of the local might improve things